src/lib.rs

```diff
@@ -1882,6 +1882,33 @@ pub trait Itertools : Iterator {
         self.all(move |elt| used.insert(elt))
     }
 
+    /// Check whether all elements compare equal, and return `Some` with the equal element.
+    ///
+    /// Empty iterators return `None`, and so do iterators with non-equal elements.
+    ///
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let data = vec![1u8, 1, 1, 2, 2, 3, 3, 3, 4, 5, 5];
+    /// assert_eq!(data.iter().equal(), None);
+    /// assert_eq!(data[0..3].iter().equal(), Some(&1));
+    /// assert_eq!(data[3..5].iter().equal(), Some(&2));
+    /// assert_eq!(data[5..8].iter().equal(), Some(&3));
+    ///
+    /// let data : Option<usize> = None;
+    /// assert_eq!(data.into_iter().equal(), None);
+    /// ```
+    fn equal(&mut self) -> Option<Self::Item>
+    where
+        Self: Sized,
+        Self::Item: PartialEq,
```

Comment:
Trait bounds in line with `all_equal` - nice.

Reply:
And now it's also renamed to `all_equal_item()`, which I love as it is discoverable when coming from `all_equal()`. See 426f0d6feba0a0af7f43501518d1bc70d47f7595 .

Reply:
Indeed, should `all_equal` be implemented as `self.all_equal_item().is_some()` now?

Reply:
A great catch, implemented in 3b1d4223c245aae0abe8303beaa964cbce816fa3 .